services/handlers/handlers.py

```diff
@@ -316,3 +317,41 @@ def print_name(value: str = None) -> str:
     Возвращает введенный текст.
     """
     return value
+
+
+@route("search-bd")
+@input_error
+def search_birthday_boy(days: str) -> Optional[str]:
+    """
+    По этой команде бат выводит в консоль все контакты, у которых день рождения через заданное количество дней.
+    Пользователь вводит команду search-birthday и количество дней, обязательно через пробел.
+    Пример команды: search-bd 520
+    """
+    try:
+        days = int(days)
+    except ValueError:
+        raise ValueError("Можно вводить только целые числа.")
+
+    format_contacts = f"{'Name':<10} : {'Address':^15} : {'Email':^10} : {'Birthday':^10} : {'Phones':^12}\n"
+    contacts = ADDRESS_BOOK().get_all_contacts()
+    today = date.today()
+    delta = timedelta(days=days)
+    target_date = today + delta
+    found_records = 0
+
+    for contact in contacts:
+        if contact.birthday:
+            birthday = contact.birthday.value
```

Comment:
Чтобы сократить вложенность, лучше сделать
```suggestion
        if not contact.birthday:
            continue
```